.travis.yml

```diff
@@ -3,25 +3,44 @@
 # All rights reserved.
 # END COPYRIGHT BLOCK
 
+language: java
+
 services:
   - docker
 
-env:
-  - FEDORA=27
-  - FEDORA=28
+stages:
+  - validate
+  - build_rpm
+
+jobs:
+  include:
+    # Validate style of scripts
+    - stage: validate
+      install:
+        - docker pull registry.fedoraproject.org/fedora:28
+        - docker run --name=validate --detach -i -v $(pwd):/root/jss registry.fedoraproject.org/fedora:28
+        - docker exec validate dnf install -y perl ShellCheck
+      script:
+        - docker exec validate /root/jss/tools/test_perl_style.sh /root/jss
+        - docker exec validate /root/jss/tools/test_shell_style.sh /root/jss
 
-install:
-  - docker pull registry.fedoraproject.org/fedora:$FEDORA
-  - docker run
-      --name=container
-      --detach
-      -i
-      -v $(pwd):/root/jss
-      registry.fedoraproject.org/fedora:$FEDORA
-  - docker exec container dnf install -y dnf-plugins-core gcc make rpm-build
-  - docker exec container dnf copr -y enable ${JSS_4_5_REPO:-@pki/10.6}
-  - docker exec container dnf builddep -y --spec /root/jss/jss.spec
-  - docker exec container /root/jss/build.sh --with-timestamp --with-commit-id rpm
+    # Build RPMs on Fedora
+    - stage: build_rpm
+      env:
+        - FEDORA=27
+        - FEDORA=28
```

Comment:
So it looks like this approach doesn't work as it first sets F27 and then F28 and then runs the build (i.e., for the last one set which is F28)... So that's sad. :/